src/lib/datepicker/datepicker-input.ts

```diff
@@ -92,8 +91,7 @@ export class MatDatepickerInputEvent<D> {
   },
   exportAs: 'matDatepickerInput',
 })
-export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAccessor, OnDestroy,
-    Validator {
+export class MatDatepickerInputCommon<D> implements ControlValueAccessor, OnDestroy, Validator {
```

Comment:
Let's call this `MatDatepickerInputBase<T>` (The `T` will be either `D` or `DateRange<D>` in the subclasses):

```ts
class MatDatepickerInput<D> extends MatDatepickerInputBase<D> {...}
class MatDatepickerRangeInput<D> extends MatDatepickerInputBase<DateRange<D>> {...}
```